### PR TITLE
kubeadm: increase ut for certs/certlist

### PR DESCRIPTION
**Pull request by my-git9**
Increase ut for app/phases/certs

#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/phases/certs/certlist_test.go [hunk 1]
```diff
@@ -222,3 +223,110 @@ func parseCertAndKey(basePath string, t *testing.T) (*x509.Certificate, crypto.P
...
+						},
+					},
+				},
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := createKeyAndCSR(tt.args.kubeadmConfig, tt.args.cert); (err != nil) != tt.wantErr {
```
[the remark this answers is not in the document]
Comment (my-git9) [reply cmd/kubeadm/app/phases/certs/certlist_test.go:327]: Thanks for review, in this ut, only the case of "key and CSR are not exist" will create the key normally, and the other cases will exit before creation.

File: cmd/kubeadm/app/phases/certs/certlist_test.go [hunk 2]
```diff
@@ -222,3 +223,110 @@ func parseCertAndKey(basePath string, t *testing.T) (*x509.Certificate, crypto.P
...
+					ClusterConfiguration: kubeadmapi.ClusterConfiguration{
+						CertificatesDir: dir,
+					},
+				},
+				cert: nil,
+			},
+			wantErr: true,
+		},
+		{
+			name: "key and CSR are not exist",
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/certs/certlist_test.go:277]:
```suggestion
			name: "key and CSR do not exist",
```

File: cmd/kubeadm/app/phases/certs/certlist_test.go [hunk 3]
```diff
@@ -222,3 +223,110 @@ func parseCertAndKey(basePath string, t *testing.T) (*x509.Certificate, crypto.P
...
+						Config: certutil.Config{
+							CommonName: "kubernetes",
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "key or CSR are already exist",
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/certs/certlist_test.go:301]:
```suggestion
			name: "key or CSR already exist",
```

File: cmd/kubeadm/app/phases/certs/certlist_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (neolit123) [reply cmd/kubeadm/app/phases/certs/certlist_test.go:327]: ok, understood.

**Review by neolit123 — COMMENTED** [review]
PTAL at the small comments i added.

/approve
/cc @SataQiu

File: cmd/kubeadm/app/phases/certs/certlist_test.go [hunk 4]
```diff
@@ -222,3 +223,110 @@ func parseCertAndKey(basePath string, t *testing.T) (*x509.Certificate, crypto.P
 
 	return parsedCert, certPair.PrivateKey
 }
+
+func Test_createKeyAndCSR(t *testing.T) {
```
Comment (SataQiu) [inline cmd/kubeadm/app/phases/certs/certlist_test.go:227]:
```suggestion
func TestCreateKeyAndCSR(t *testing.T) {
```
What are the rules for naming test functions?
There is no use of `Test_xxx` in the kubeadm code base.

File: cmd/kubeadm/app/phases/certs/certlist_test.go [hunk 5]
```diff
@@ -222,3 +223,110 @@ func parseCertAndKey(basePath string, t *testing.T) (*x509.Certificate, crypto.P
...
+			args: args{
+				kubeadmConfig: &kubeadmapi.InitConfiguration{
+					NodeRegistration: kubeadmapi.NodeRegistrationOptions{
+						Name: "test-node",
+					},
+					ClusterConfiguration: kubeadmapi.ClusterConfiguration{
+						CertificatesDir: dir,
+					},
+				},
+				cert: &KubeadmCert{
```
Comment (SataQiu) [inline cmd/kubeadm/app/phases/certs/certlist_test.go:287]: There are many duplicate definitions of `kubeadmConfig` and `KubeadmCert`.
We can define them as variables (`validKubeadmConfig` and `validKubeadmCert` ?) and reference them in the test cases.
WDYT? @my-git9

**Comment by SataQiu** [comment]
thanks @my-git9 
/lgtm
/approve

**Pull request merged**